ci/build-and-test-nupic-bindings.sh

```diff
@@ -0,0 +1,171 @@
+   artifacts from prior build attempts. Otherwise, behavior is undefined.
+
+
+INPUT ENVIRONMENT VARIABLES:
+
+  BUILD_TYPE : Specifies build type, which may be either Release or Debug;
+               defaults to Release. [OPTIONAL]
+  RESULT_KEY : Build result key; used to decorate artifact names. [OPTIONAL]
+  WHEEL_PLAT : Wheel platform name; pass manylinux1_x86_64 for manylinux build;
+               leave undefined for all other builds.
+
+OUTPUTS:
+  nupic.bindings wheel: On success, the resulting wheel will be located in the
+                        subdirectory nupic_bindings_wheelhouse of the source
+                        tree's root directory.
+
+  test results: nupic.bindings test results will be located in the subdirectory
+                test_results of the source tree's root directory with the
+                the following content:
+
+                cplusplus-test-results.txt
+                junit-test-results.xml
+                htmlcov-report/
+
+"
+
+if [[ $1 == --help ]]; then
+  echo "${USAGE}"
+  exit 0
+fi
+
+if [[ $# > 0 ]]; then
+  echo "ERROR Unexpected arguments: ${@}" >&2
+  echo "${USAGE}" >&2
+  exit 1
+fi
+
+
+set -o xtrace
+
+
+# Apply defaults
+BUILD_TYPE=${BUILD_TYPE-"Release"}
+
+
+NUPIC_CORE_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+DEST_WHEELHOUSE="${NUPIC_CORE_ROOT}/nupic_bindings_wheelhouse"
+
+TEST_RESULTS_DIR="${NUPIC_CORE_ROOT}/test_results"
+
+echo "RUNNING NUPIC BINDINGS BUILD: BUILD_TYPE=${BUILD_TYPE}, " \
+     "RESULT_KEY=${RESULT_KEY}, DEST_WHEELHOUSE=${DEST_WHEELHOUSE}" >&2
+
+# Install pycapnp to get the matching capnproto headers for nupic.core build
+# NOTE Conditional pycapnp dependency should be incorporated into
+# bindings/py/requirements.txt to abstract it from upstream scripts.
+pip install pycapnp==0.5.8
+
+# Install nupic.bindings dependencies; the nupic.core cmake build depends on
+# some of them (e.g., numpy).
+pip install -r ${NUPIC_CORE_ROOT}/bindings/py/requirements.txt
+
+
+#
+# Build nupic.bindings
+#
+
+# ZZZ debug statement to see if build/scripts was inherited from another job
+ls ${NUPIC_CORE_ROOT}/build/scripts || true
+# ZZZ end debug
+
+mkdir -p ${NUPIC_CORE_ROOT}/build/scripts
+cd ${NUPIC_CORE_ROOT}/build/scripts
+
+# Configure nupic.core build
+if [[ "$BUILD_TYPE" == "Debug" ]]; then
+  EXTRA_CMAKE_DEFINITIONS="-DNUPIC_IWYU=ON -DNTA_COV_ENABLED=ON"
+fi
+
+cmake ${NUPIC_CORE_ROOT} \
+    -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
+    ${EXTRA_CMAKE_DEFINITIONS} \
+    -DCMAKE_INSTALL_PREFIX=${NUPIC_CORE_ROOT}/build/release \
+    -DPY_EXTENSIONS_DIR=${NUPIC_CORE_ROOT}/bindings/py/nupic/bindings
+
+# Build nupic.core
+make install
+
+# Build nupic.bindings python extensions from nupic.core build artifacts
+if [[ $WHEEL_PLAT ]]; then
+  EXTRA_WHEEL_OPTIONS="--plat-name ${WHEEL_PLAT}"
+fi
+
+cd ${NUPIC_CORE_ROOT}
+python setup.py bdist_wheel --dist-dir ${DEST_WHEELHOUSE} ${EXTRA_WHEEL_OPTIONS}
```

Comment:
Move this, and the installation of the wheel, after running the C++ binaries are executed, right before you test the Python installation. Keeps the C++ and Python side separate.